shared/js/background/declarative-net-request.js

```diff
@@ -25,7 +26,8 @@ export const SETTING_PREFIX = 'declarative_net_request-'
 // of removing rules associated with different configurations.
 const ruleIdRangeByConfigName = {
     tds: [1, 10000],
-    config: [10001, 20000]
+    config: [10001, 20000],
```

Comment:
I know these changes are subject to change, but currently the new rule range will clash with the constant IDs (e.g. `USER_ALLOWLIST_RULE_ID`) below. This seems like a "foot-gun waiting to happen" (my bad), so how about adding a "_RESERVED" range of 1 to 1000, and tweaking those constant IDs to be within that?
```javascript
const ruleIdRangeByConfigName = {
    _RESERVED: [1, 1000],
    tds: [1001, 11000],
    config: [11001, 21000],
    combined: [21001, 30000]
}
```

Reply:
> and tweaking those constant IDs to be within that?

You mean the allowlist IDs should be in 1-1000?